### DISCUSSION TITLE
Current state of integration

### DISCUSSION THREAD

**Discussion by stefanke** [sven819/SharpClimateUART-ESPHome#5] [General]
I plan to add a Bosch 8000i to our house. Your integration seems to be the only one which (possibly) enables the connection to home assistant. Before buying the unit I wanted to ask about the current state of development and possible problems. 

I understand this is an early version, but please allow some questions:
- Is it possible to control all settings which can be controlled via the remote through home assistant?
- Are there any major problems or topics already known that could block the development of this approach?

For some reasons the 8000i is the best solution for us, but without home assistant integration I would need to research again. 
thanks for your help,
Stefan

**Comment by sven819** [comment]
I have now been running it continuously, and I am not aware of any further issues. All functions are available, just like with the original Wi-Fi module, except for the timer.
Please note that there is more than one version of the 8000. The 8000 has been tested and works, but as far as I can tell, the 8001 has not yet been tested.
Maybe it’s worth asking in the ESPHome forum if anyone is using this exact variant.

**Comment by tremblay0934** [comment]
Hi @sven819 , in the same line of question, do you know if it is possible to interface the unit (sharp in my case) to extract the statistics data? 

The Sharp Air app showcases some daily wat usage. I would assume the only data sampled here by the OEM microcontroller is some kind ADC raw voltage, which is sampled periodically and the SHARP cloud app buffers the raw data and showcases graphs.

If this is on your roadmap and you have some kind of idea on how to reverse engineer those specific frames, I could definitely help.

Thank you!

**Comment by sven819** [comment]
Hi,
that’s interesting. I’m looking for this data. Can you record the communication?
Only the current consumption is important; the rest would be handled by Home Assistant.

**Comment by tremblay0934** [comment]
> Can you record the communication?

You mean probe the tx/Rx line and send you a screenshot so that you can decode?

**Comment by sven819** [comment]
> > Can you record the communication?
> 
> You mean probe the tx/Rx line and send you a screenshot so that you can decode?

Please log the data in hex format with the initial connection. Then we can see if something can be done for Sharp, or even for both Sharp and Bosch/Buderus.
